common/overloads/overloads.go

```diff
@@ -271,3 +271,22 @@ const (
 	HasNext  = "@hasNext"
 	Next     = "@next"
 )
+
+// IsTypeConversionFunction returns whether the input function is a standard library type
+// conversion function.
+func IsTypeConversionFunction(function string) bool {
+	switch function {
+	case TypeConvertBool,
+		TypeConvertBytes,
+		TypeConvertDouble,
+		TypeConvertDuration,
+		TypeConvertInt,
+		TypeConvertString,
+		TypeConvertTimestamp,
+		TypeConvertType,
+		TypeConvertUint:
```

Comment:
Consider whether TypeConvertDyn should also give a positive answer. If not, add a comment to explain the discrepancy versus the list of `TypeConvert*` constants above.

Note that the language spec lacks type conversion functions for `bool()` and `type()`, while it contains `list()` and `map()` conversions. Should the former be added to the spec?  Should the latter be added to the list here?  Looks like we implement a string-to-bool conversion that's not in the spec.

Reply:
It looks like `list()` and `map()` type conversions were never implemented. I'm actually having a hard time making sense of what they were intended to do. :/ They sort of look like identity functions:

```
list(T) -> list(T) // ???
```

With respect to the places where cel-go goes beyond the spec, the type conversions are usually pretty reasonable. I may have picked a strange name for `type(uint)` as `TypeConvertType`, but it's the standard type function: `type(uint) == type`, `type(null_type) == type`. Stuff like that.

Reply:
Per https://github.com/google/cel-spec/issues/123, I'll forego the `list()` and `map()` conversion functions, but will add `dyn()`. 

With respect to `bool()` I believe the identity function should be supported and I'm not opposed to supporting `string(bool("true")) == "true"` since this seems like reasonable round-tripping behavior even if its infrequently used.